### ISSUE TITLE
No skills found Toast message appears multiple times

### ISSUE THREAD

**Issue by ycwei982** [fossasia/susi_android#1782]
**Actual Behaviour**

Searching for something in *SUSI.AI Skills*.

**Expected Behaviour**

It should pop `No skill found yet` when I press Enter.

**Steps to reproduce it**

1. Go to **SUSI.AI Skills*.
2. Press the search button.
3. Type something.

**LogCat for the issue**

No crash happens so I think it is not necessary

**Screenshots of the issue**

![ezgif-2-4dc7903ed76e](https://user-images.githubusercontent.com/11908157/48897805-06170380-ee86-11e8-963e-0ae544fe8feb.gif)

**Would you like to work on the issue?**

I will try but not 100% I can

**Comment by muarachmann** [comment]
@ycwei982 i think they are listening to the ontext change listener so in my opinion you must not wait to press the enter button before you search a skill 😃

**Comment by ycwei982** [comment]
@muarachmann 

As GIF showed, I did not press Enter (Search) button on my keyboard.

I also inspected code, and I found it won't listen to `IME_ACTION_SEARCH`, it will only listen the text change and search the strings every time as soon as the EditText view content changes.

**Comment by NileshPranami** [comment]
@ycwei982  are you still working on this issue?
If not I wanna go ahead with it.

**Comment by ycwei982** [comment]
Sorry I'm not working on it now. I'm not familiar with Kotlin Android.

**Comment by NileshPranami** [comment]
@iamareebjamal Sir, compiler can't go inside the first for loop given below. I can't get it why. Please help me out in this. Sample code is of SkillActivity.kt. Is there is any condition error.
```
fun performSearch(query: String): Boolean {

        for ((pos, item) in skills.withIndex()) {
            if (query in item.first) {
                skillMetrics.scrollToPosition(pos)
                return true
            }

            for (item2 in item.second) {
                if (query.toLowerCase() in item2.group.toLowerCase()) {
                    skillMetrics.scrollToPosition(pos)
                    return true
                }
            }
        }
        Toast.makeText(this, R.string.skill_not_found, Toast.LENGTH_SHORT).show()
        return false
    }

```

**Comment by iamareebjamal** [comment]
Check what is returned in `skills.withIndex()`

**Comment by NileshPranami** [comment]
@iamareebjamal Sir, actually here skills variable is only declared but doesn't assign any value  and on print it show this output kotlin.collections.IndexingIterable@8bce247.
How we can assign the values in skills since its datatype is ArrayList<Pair<String, List<SkillData>>>

**Comment by iamareebjamal** [comment]
@arundhati24 @batbrain7 Over to you

**Comment by NileshPranami** [comment]
@batbrain7 @iamareebjamal 
I need help in this context.
I have tried alternative methods but that didn't work. I have tried    
 private var skills1= GroupWiseSkills("",ArrayList())
instead of ArrayList<Pair<String, List>> datatype but what I got problem is skills in SkillActivity is not get data.

error message on printing name on index 1
IndexOutOfBoundsException: Index: 1, Size: 0

I have to complete this anyhow so please help me.

**Comment by atm1504** [comment]
@iamareebjamal @batbrain7 @arundhati24  is this issue solved? If not can I start working on it?

**Comment by immadisairaj** [comment]
@iamareebjamal I think this issue can be closed as it is already fixed in #2095
